src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -115,17 +118,39 @@ def set_query_compiler(
         copy: bool = False,
         name: object = None,
         tupleize_cols: bool = True,
+        convert_to_index: bool = True,
     ) -> None:
         """
         Helper method to find and save query compiler when index should be lazy
         """
+        from snowflake.snowpark.modin.pandas import Series
         from snowflake.snowpark.modin.pandas.dataframe import DataFrame
         from snowflake.snowpark.modin.plugin.compiler.snowflake_query_compiler import (
             SnowflakeQueryCompiler,
         )
 
         if isinstance(data, SnowflakeQueryCompiler):
             qc = data
+            if name is not None:
+                qc = qc.set_index_names([name])
+        elif isinstance(data, (DataFrame, Series)):
+            if convert_to_index:
+                qc = data._query_compiler
+                # TODO: move Series data columns to Index index columns
+                # old_frame = data._query_compiler._modin_frame
+                # from snowflake.snowpark.modin.plugin._internal.frame import InternalFrame
+                # new_frame = InternalFrame.create(
+                #     ordered_dataframe=old_frame.ordered_dataframe,
+                #     data_column_pandas_labels=old_frame.data_column_pandas_labels,
+                #     data_column_pandas_index_names=old_frame.data_column_pandas_index_names,
+                #     data_column_snowflake_quoted_identifiers=old_frame.data_column_snowflake_quoted_identifiers,
+                #     index_column_pandas_labels=old_frame.data_column_pandas_labels,
+                #     index_column_snowflake_quoted_identifiers=old_frame.data_column_snowflake_quoted_identifiers,
+                # )
+                # qc = SnowflakeQueryCompiler(new_frame)
```

Comment:
did you mean to leave this commented out?

Reply:
Yep, it doesn't work. I need to create a ticket to explain what I was trying to do there, but I can delete it if thats better